app/main.py

```diff
@@ -1,25 +1,81 @@
-from abc import ABC
+from abc import ABC, abstractmethod
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: Any) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: type) -> Any:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: Any) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"{self.protected_name} out of range "
+                             f"[{self.min_amount}-{self.max_amount}]")
+        setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(
+            self,
+            age_range: tuple,
+            weight_range: tuple,
+            height_range: tuple
+    ) -> None:
+        self.age_range = age_range
+        self.weight_range = weight_range
+        self.height_range = height_range
+
+    @abstractmethod
+    def can_access(self, visitor: Visitor) -> None:
+        pass
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    def __init__(self) -> None:
+        super().__init__((4, 14), (20, 50), (80, 120))
+
+    def can_access(self, visitor: Visitor) -> bool:
+        return (
+            self.age_range[0] <= visitor.age <= self.age_range[1]
+            and self.weight_range[0] <= visitor.weight <= self.weight_range[1]
+            and self.height_range[0] <= visitor.height <= self.height_range[1]
+        )
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    def __init__(self) -> None:
+        super().__init__((14, 60), (50, 120), (120, 220))
+
+    def can_access(self, visitor: Visitor) -> bool:
+        return (
+            self.age_range[0] <= visitor.age <= self.age_range[1]
+            and self.weight_range[0] <= visitor.weight <= self.weight_range[1]
+            and self.height_range[0] <= visitor.height <= self.height_range[1]
+        )
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class: SlideLimitationValidator
```

Comment:
Change your type annotation to indicate that you expect to obtain subclasses of the `SlideLimitationValidator` class for `limitation_class`, not `SlideLimitationValidator` instances.